tools/cldr-apps/src/main/java/org/unicode/cldr/web/ClaGithubList.java

```diff
@@ -0,0 +1,158 @@
+         */
+        public Date getSignedAt() {
+            if (signed_at == null || signed_at.isEmpty()) return null;
+            return Date.from(Instant.parse(signed_at));
+        }
+    }
+
+    static final Logger logger = SurveyLog.forClass(ClaGithubList.class);
+
+    public static final ClaGithubList getInstance() {
+        return Helper.INSTANCE;
+    }
+
+    private static final class Helper {
+
+        static final ClaGithubList INSTANCE = new ClaGithubList();
+    }
+
+    static final Gson gson = new Gson();
+
+    private final CLDRConfig instance = CLDRConfig.getInstance();
+
+    private final String CLA_FILE = instance.getProperty("CLA_FILE", "./signatures.json");
+    private final Path claFilePath;
+
+    ClaGithubList() {
+        final File homeFile = ((CLDRConfigImpl) instance).getHomeFile();
+        claFilePath = new File(homeFile, CLA_FILE).toPath();
+        logger.info("CLA_FILE=" + claFilePath.toString());
+    }
+
+    public SignEntry getSignEntry(final String id) {
+        // Impl:  reread each time
+        Map<String, SignEntry> allSigners = new HashMap<>();
+        try (final Reader r = new FileReader(claFilePath.toFile(), StandardCharsets.UTF_8);
+                final JsonReader jr = gson.newJsonReader(r); ) {
+            jr.beginArray();
+            while (jr.hasNext()) {
+                SignEntry e = new SignEntry();
+                jr.beginObject();
+                while (jr.hasNext()) {
+                    switch (jr.nextName()) {
+                        case "user_name":
+                            e.user_name = jr.nextString();
+                            break;
+                        case "signed_at":
+                            e.signed_at = jr.nextString();
+                            break;
+                        case "revoked_at":
+                            e.revoked_at = jr.nextString();
+                            break;
+                        case "name":
+                            e.name = jr.nextString();
+                            break;
+                        case "category":
+                            e.category = jr.nextString();
+                            break;
+                        case "employer":
+                            e.employer = jr.nextString();
+                            break;
+                        case "email":
+                            e.email = jr.nextString();
+                            break;
+                        default:
+                            jr.skipValue();
+                            break; // ignore
+                    }
+                }
+                jr.endObject();
+
+                // the list is in REVERSE order (latest first). So we only track the first
+                // appearance.
+                if (!allSigners.containsKey(e.user_name)) {
+                    allSigners.put(e.user_name, e);
+                    // TODO: we could break here.
```

Comment:
again, TODO ideally has ticket number -- but in this case it's not clear whether anything needs to be done; what happens if we don't break here?

Reply:
It means that it's already found the data item that the only current caller of this function cares about, so if it doesn't break here, it will perform unnecessary processing.

Reply:
so, `break`?

Reply:
I'd actually rather not break here because it makes the API harder to follow. (it would be `if user.name.equals(id)` anyway).